openff/qcsubmit/_tests/results/test_results.py

```diff
@@ -381,6 +381,59 @@ def test_optimization_create_basic_dataset(optimization_result_collection):
     assert dataset.n_records == 5  # the collection contains 1 duplicate
 
 
+def test_optimization_create_basic_dataset_297():
+    """Test creating a new ``BasicDataset`` from the result of an optimization
+    dataset, and verify that the molecule hashes match to prevent the creation
+    of separate records on QCArchive. See issue #297 for more details.
+    """
+
+    # these are two real entries from the "OpenFF Sulfur Optimization Training
+    # Coverage Supplement v1.0" dataset used in issue #297. the first fails the
+    # round-trip in the previous create_basic_dataset implemenation but the
+    # second works with either
+    opt = OptimizationResultCollection.parse_raw(
+        """
+        {
+            "entries": {
+                "https://api.qcarchive.molssi.org:443/": [
+                {
+                    "type": "optimization",
+                    "record_id": 138340682,
+                    "cmiles": "[H:11][c:10]1[c:12]([c:14]([c:16]2[c:7]([c:8]1[H:9])[C:5](=[O:6])[C:3](=[C:1]([S:17]2(=[O:18])=[O:19])[H:2])[Br:4])[H:15])[H:13]",
+                    "inchi_key": "UKQYACGPJCUZJN-UHFFFAOYNA-N"
+                },
+                {
+                    "type": "optimization",
+                    "record_id": 138341382,
+                    "cmiles": "[H:18][C@@:10]([C:11](=[O:12])[O:13][H:14])([C:9]([H:19])([H:20])[C:8]([H:21])([H:22])[S:2](=[N:1][H:23])(=[O:3])[C:4]([H:5])([H:6])[H:7])[N:15]([H:16])[H:17]",
+                    "inchi_key": "SXTAYKAGBXMACB-AIXGRKOHNA-N"
+                }
+                ]
+            },
+            "provenance": {},
+            "type": "OptimizationResultCollection"
+        }
+        """
+    )
+
+    opt_hashes = {rec.final_molecule.get_hash() for rec, _mol in opt.to_records()}
+
+    basic = opt.create_basic_dataset(
+        "dummy basic dataset name",
+        "descdesc",
+        "tagtagtag",
+        driver="hessian",
+    )
+
+    bas_hashes = {mol.molecule.get_hash() for mol in basic._get_entries()}
```

Comment:
(blocking) I recall that about 1/3rd of the original dataset still had an identical hash, even after whatever numerical mangling had happened. Given that those molecules had tens of atoms with coordinates, any one of which could have triggered the coordinate rounding error, and it still only happened 1/3rd of the time, this should use an input that we know will trigger the error instead of relying on a molecule as small as methane (with only 5 coords) and the rdkit-backed `offmol.generate_conformers` method (which could change behavior when using a different version of rdkit).

Reply:
Resolving because this wasn't using methane coords at all, THIS test was actually reaching out to real QCA to get real confs (and in particular, ones that BW established DO have the bad behavior in question)